app/views/todos/edit/[id].tsx

```diff
@@ -1,32 +1,126 @@
+import { usePowerSync, usePowerSyncWatchedQuery } from '@journeyapps/powersync-sdk-react-native';
+import { CameraCapturedPicture } from 'expo-camera';
+import _ from 'lodash';
 import * as React from 'react';
 import { StatusBar } from 'expo-status-bar';
 import { ScrollView, View, Text } from 'react-native';
 import { FAB } from 'react-native-elements';
-import { observer } from 'mobx-react-lite';
 import { Stack, useLocalSearchParams } from 'expo-router';
 import prompt from 'react-native-prompt-android';
-
-import { useSystem } from '../../../../library/stores/system';
+import {
+  ATTACHMENT_TABLE,
+  AttachmentRecord,
+  TODO_TABLE,
+  TodoRecord,
+  LIST_TABLE
+} from '../../../../library/powersync/AppSchema';
+import { useSystem } from '../../../../library/powersync/system';
 import { TodoItemWidget } from '../../../../library/widgets/TodoItemWidget';
 
-const TodoView = observer(() => {
-  const { listStore, todoStore } = useSystem();
+type TodoEntry = TodoRecord & Partial<Omit<AttachmentRecord, 'id'>> & { todo_id: string; attachment_id: string | null };
+
+const toAttachmentRecord = _.memoize((entry: TodoEntry): AttachmentRecord | null => {
+  return entry.attachment_id == null
+    ? null
+    : {
+        id: entry.attachment_id,
+        filename: entry.filename!,
+        state: entry.state!,
+        timestamp: entry.timestamp,
+        local_uri: entry.local_uri,
+        media_type: entry.media_type,
+        size: entry.size
+      };
+});
+
+const TodoView: React.FC = () => {
+  const system = useSystem();
+  const powerSync = usePowerSync();
   const params = useLocalSearchParams<{ id: string }>();
+  const listID = params.id;
 
-  const id = params.id;
-  const listModel = React.useMemo(() => {
-    if (!id) {
-      return null;
+  const [listRecord] = usePowerSyncWatchedQuery<{ name: string }>(`SELECT name FROM ${LIST_TABLE} WHERE id = ?`, [
+    listID
+  ]);
+
+  const todos = usePowerSyncWatchedQuery<TodoEntry>(
+    `
+        SELECT
+            ${TODO_TABLE}.id AS todo_id,
+            ${TODO_TABLE}.*,
+            ${ATTACHMENT_TABLE}.id AS attachment_id, 
+            ${ATTACHMENT_TABLE}.*
+        FROM 
+            ${TODO_TABLE}
+        LEFT JOIN 
+            ${LIST_TABLE} ON ${TODO_TABLE}.list_id = ${LIST_TABLE}.id
+        LEFT JOIN 
+            ${ATTACHMENT_TABLE} ON ${TODO_TABLE}.photo_id = ${ATTACHMENT_TABLE}.id
+        WHERE 
+            ${TODO_TABLE}.list_id = ?`,
+    [listID],
+    // NOTE: It is important to add the tables to the watch list, otherwise the query will not be re-executed
+    { tables: [TODO_TABLE, ATTACHMENT_TABLE, LIST_TABLE] }
```

Comment:
This is okay for now, but should be fixed in the underlying lib. I assume it's just this bit that doesn't work for local-only tables:
https://github.com/journeyapps/powersync-react-native-sdk/blob/af0031b0eabb4524ad2a507b09487c68d268abb1/packages/powersync-sdk-common/src/client/AbstractPowerSyncDatabase.ts#L404C7-L404C88